app/assets/javascripts/app/load_cart_count.js

```diff
@@ -0,0 +1,35 @@
+window.CartCount = class CartCount {
+  constructor($cart) {
+    this.$cart = $cart;
+  }
+
+  loadCartCount() {
+    const element = $(this.$cart);
+
+    if (!element) { return; }
+
+    let url = element.data("url");
+
+    if (!url) { return; }
+
+    url = new URL(url);
+
+    fetch(url, { mode: "no-cors" }).then(function (response) {
+      if (response.ok) {
+        response.text().then(function (data) {
+          data = JSON.parse(data);
+          const count = data.data.count;
+          const text = `Cart (${count})`;
+
+          const anchorElement = element.find("a");
+          anchorElement.text(text);
+        });
+      }
+    });
+  }
+};
+
+$(function () {
```

Comment:
This is fine with me, but I might consider something simpler to start.
I don't feel strongly about this but I would just define a function like `updateCartCount` that takes a string for the class selector - or just hard code `".js--cart_count")` to start.
Then when the page is done loading, run `updateCartCount`.  

What you have here works and there's nothing wrong with using a constructor, totally up to you.